pkg/util/docker/dockerfile/builder/client.go

```diff
@@ -51,6 +54,8 @@ type ClientExecutor struct {
 	AuthFn func(name string) ([]docker.AuthConfiguration, bool)
 	// HostConfig is used to start the container (if necessary).
 	HostConfig *docker.HostConfig
+	// LogFn is an optional command to log information to the end user
+	LogFn func(format string, args ...interface{})
```

Comment:
We have AuthFn already, and defining functions specifically as Fn is fairly common for us.